src/main/java/ch/njol/skript/expressions/ExprBlockSound.java

```diff
@@ -92,11 +98,12 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 
 	@Override
 	protected String @Nullable [] get(Event event) {
+		//noinspection rawtypes,deprecation
 		return objects.stream(event)
 			.map(this::convertAndGetSound)
 			.filter(Objects::nonNull)
 			.distinct()
-			.map(Sound::name)
+			.map(SOUND_IS_INTERFACE ? (sound -> sound.getKey().getKey()) : (sound -> ((Enum) sound).name()))
```

Comment:
For consistency sake, why not just use the key for all versions, rather than key on 1.21.3+ and enum name on versions below?

Reply:
oh i didn't realize that was available on previous versions

Reply:
to the best of my knowledge it was added around 1.16.x (since "feature" is for MC 1.19+, should be fine)

Reply:
yeah checked helpchat 1.19.4, it's there